org.eclipse.xtext.tests/src/org/eclipse/xtext/formatting2/partialformatting/PartialFormattingTest.xtend

```diff
@@ -0,0 +1,113 @@
+/*******************************************************************************
+ * Copyright (c) 2017 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.formatting2.partialformatting
+
+import com.google.inject.Inject
+import org.eclipse.xtext.testing.InjectWith
+import org.eclipse.xtext.testing.XtextRunner
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.eclipse.xtext.formatting2.internal.GenericFormatterTester
+
+import org.eclipse.xtext.util.TextRegion
+import org.eclipse.xtext.formatting2.internal.GenericFormatter
+import org.eclipse.xtext.formatting2.regionaccess.ITextRegionExtensions
+import org.eclipse.xtext.formatting2.IFormattableDocument
+import org.eclipse.xtext.testlanguages.backtracking.beeLangTestLanguage.Model
+import org.eclipse.xtext.testlanguages.backtracking.beeLangTestLanguage.Unit
+
+/**
+ * @author cwehrheim - Initial contribution and API
+ */
+@RunWith(XtextRunner)
+@InjectWith(MyPartialFormattingTestLanguageInjectorProvider)
+class PartialFormattingTest {
+	@Inject extension GenericFormatterTester
+
+	@Test
+	def void testFormatting1() {
```

Comment:
This is the test that is already working without your change, isn't it? I think we don't need this one.

Instead, there should be a test for the second scenario you're implementing: TextReplacements overlapping `allowedRegion.endOffset`.